action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -406,6 +426,25 @@ func (stateDB *StateDBAdapter) Suicide(evmAddr common.Address) bool {
 		stateDB.logError(err)
 		return false
 	}
+	// before calling Suicide, EVM will transfer the contract's balance to beneficiary
+	// need to create a transaction log on successful suicide
+	if stateDB.lastAddBalanceAmount.Cmp(actBalance) == 0 {
+		if len(stateDB.lastAddBalanceAmount.Bytes()) > 0 {
+			from, _ := address.FromBytes(evmAddr[:])
+			stateDB.addTransactionLogs(&action.TransactionLog{
+				Type:      iotextypes.TransactionLogType_IN_CONTRACT_TRANSFER,
+				Sender:    from.String(),
+				Recipient: stateDB.lastAddBalanceAddr,
+				Amount:    stateDB.lastAddBalanceAmount,
+			})
+		}
+	} else {
+		if stateDB.suicideTxLogMismatchPanic {
```

Comment:
for safety, we gate the panic with a flag
after hard-fork, we can run a fullsync w/o the flag, and remove this flag if fullsync pass

Reply:
https://github.com/iotexproject/iotex-core/pull/4171/files#r1519122441 if we panic here, what's the purpose of resetting the mount?

Reply:
I gave an example above, if we don't reset the amount, it is possible to generate a mistaken txLog for an earlier `AddBalance()` whose amount happens to equal to suicide address